scripts/sql/hssc_obs_fact_etl.sql

```diff
@@ -0,0 +1,404 @@
+             end_date, modifier_cd,instance_num,valtype_cd,tval_char,
+             valueflag_cd,units_cd, location_cd,import_date,sourcesystem_cd)
+            values (mo.visit_deid,mo.patient_deid,'RXCUI:' || mo.med_code,'@',
+            mo.shifted_start_date, mo.shifted_end_date, '@',mo.rank,
+            '@','@','@','@','@',sysdate,'CDW');
+
+          if mo.admin = 'A' then
+             insert into i2b2hsscdata.observation_fact  
+             (encounter_num,patient_num,concept_cd, provider_id,start_date,
+               end_date, modifier_cd,instance_num,valtype_cd,tval_char,
+               valueflag_cd,units_cd, location_cd,import_date,sourcesystem_cd)
+                values (mo.visit_deid,mo.patient_deid,'RXCUI:' || mo.med_code,
+                '@', mo.shifted_start_date, mo.shifted_end_date, 
+                'MED:ADMIN',mo.rank,'T',mo.admin,NULL,'@','@',sysdate,'CDW');
+          end if;
+
+          EXCEPTION WHEN OTHERS 
+          THEN
+          pkg_error.log(p_error_code => substr(sqlerrm,1,9),
+              p_error_message => substr(sqlerrm,12) || '.EXCP ' 
+              || mo.visit_deid || ', ' || mo.patient_deid,
+              p_package => '', p_procedure => m_procname);
+      
+        m_rowcnt := m_rowcnt + 1;
+
+        if ( m_rowcnt > 0 and mod(m_rowcnt, m_comrows) = 0 ) then
+              COMMIT;
+        end if;
+        END;
+        
+    end if;  
+  end loop;
+  
+  commit;
+
+END ETL_MEDICATION_ORDER;
+-------------------------------------------------------------------------------------------------
+
+
+create or replace PROCEDURE ETL_VITAL authid current_user
+IS
+
+   m_rowcnt NUMBER := 0; -- row counter
+   m_comrows NUMBER := 40000; -- commit every m_comrows
+   m_procname ERRORLOG.PROCEDURE_NAME%TYPE := 'ETL_VITAL';
+
+
+  cursor vital_cur is 
+    SELECT DISTINCT
+      /*+ PARALLEL 8 */
+      D.PATIENT_DEID AS PATIENT_NUM,
+      M.VISIT_DEID AS ENCOUNTER_NUM,
+      'VITAL:' || decode(V.OBSERVATION_TYPE,'HEIGHT','HT','WEIGHT','WT') 
+        as CONCEPT_CD,
+  
+      COALESCE(V.COLLECTION_DATE, S.VISIT_START_DATE) - 
+        D.SHIFTVALUE as START_DATE, -- modified
+      COALESCE(V.COLLECTION_DATE, S.VISIT_END_DATE) - 
+        D.SHIFTVALUE as END_DATE, -- modified
+    
+      '1' as INSTANCE_NUM, 
+      '@' as PROVIDER_ID, 
+
+      -- convert height to inches and weight to pounds 
+      -- todo look at the unit of measure (don't assume it's cm/kg)
+      ROUND(decode(v.observation_type,'HEIGHT', 
+        V.VITAL_VALUE_NUM * .3937,
+        V.VITAL_VALUE_NUM * 2.2046),2) 
+          as NVAL_NUM,  
+
+      '@' as VALUEFLAG_CD, 
+      null as QUANTITY_NUM,  
+      DECODE(V.OBSERVATION_TYPE,'HEIGHT','inches','lbs') as UNITS_CD,
+      'N' as valtype_cd,
+      'E' as tval_char---,
+      ----V.HTB_ENC_ACT_ID as ACT_ID,
+      ----V.HTB_ENC_ACT_VER_NUM AS ACT_VER_NUM
+      FROM cdw.VITAL@dtdev V
+      JOIN cdw.Patient_deid_map_hssc@dtdev D ON (V.PATIENT_ID = D.PATIENT_ID)
```

Comment:
`V.PATIENT_ID` should be `S.PATIENT_ID`. (`V` is an alias of `VITAL`; `S` is an alias of `VISIT`.)
`PATIENT_ID` on the `VITAL` table should be avoided.  It will not be on the fully integrated version of the `VITAL` data.

Consider using aliases that more descriptive:  maybe, for example, `vit` for `vital` and `enc` for `visit` (`encounter`).